src/vcUnitConversion.cpp

```diff
@@ -0,0 +1,65 @@
+#include "vcUnitConversion.h"
+
+double vcUnitConversion_ConvertDistance(double sourceValue, vcDistanceUnit sourceUnit, vcDistanceUnit requiredUnit)
+{
+  if (sourceUnit == requiredUnit)
+    return sourceValue;
+
+  const double metreTable[vcDistance_Count] = { 1.0, 1000.0, 0.01, 0.001, 1200.0 / 3937.0, 5280.0 * 1200.0 / 3937.0, 1.0 / 39.37, 1852.0 };
```

Comment:
This is much better, I was going to say it would still be an improvement to pre-calculate these values into a table, but probably not as the lookup constitutes an additional multiplication anyway